manim/config.py

```diff
@@ -1,283 +1,406 @@
+
+    # the *_quality arguments happen to override options that are all ints
+    for flag in ['fourk_quality', 'high_quality', 'medium_quality',
+                 'low_quality']:
+        if getattr(args, flag) is not None:
+            config['CAMERA_CONFIG'].update(
+                {opt.upper(): config_parser[flag].getint(opt)
+                 for opt in config_parser[flag]})
+
+    # Parse the -r (--resolution) flag.  Note the -r flag does not
+    # correspond to any section in manim.cfg, but overrides the same
+    # options as the *_quality sections.
+    if args.resolution is not None:
         if  "," in args.resolution:
             height_str, width_str = args.resolution.split(",")
             height = int(height_str)
             width = int(width_str)
         else:
             height = int(args.resolution)
             width = int(16 * height / 9)
-        camera_config.update({
-            "pixel_height": height,
-            "pixel_width": width,
-        })
+        config['CAMERA_CONFIG'].update({'pixel_height': height,
+                                        'pixel_width': width})
 
-    if args.color:
+    # Parse the -c (--color) flag
+    if args.color is not None:
         try:
-            camera_config["background_color"] = colour.Color(args.color)
+            config['CAMERA_CONFIG']['BACKGROUND_COLOR'] = colour.Color(args.color)
         except AttributeError as err:
-            logger.warning("Please use a valid color")
+            logger.warning('Please use a valid color')
             logger.error(err)
             sys.exit(2)
 
-    # If rendering a transparent image/move, make sure the
-    # scene has a background opacity of 0
-    if args.transparent:
-        camera_config["background_opacity"] = 0
-
-    return camera_config
+    # As before, make FILE_WRITER_CONFIG by pulling form the overall
+    config['FILE_WRITER_CONFIG'] = {key: config[key] for key in [
+        "WRITE_TO_MOVIE", "SAVE_LAST_FRAME", "SAVE_PNGS",
+        "SAVE_AS_GIF", "PNG_MODE", "MOVIE_FILE_EXTENSION",
+        "OUTPUT_FILE", "INPUT_FILE"]}
 
+    return config
 
-def initialize_directories(config):
-    dir_config = {}
-    dir_config["media_dir"] = config["media_dir"] or dirs.MEDIA_DIR
-    dir_config["video_dir"] = config["video_dir"] or dirs.VIDEO_DIR
 
-    if not (config["video_dir"] and config["tex_dir"]):
-        if config["media_dir"]:
-            if not os.path.isdir(dir_config["media_dir"]):
-                os.makedirs(dir_config["media_dir"])
-        if not os.path.isdir(dir_config["media_dir"]):
-            dir_config["media_dir"] = "./media"
+def _init_dirs(config):
+    if not (config["VIDEO_DIR"] and config["TEX_DIR"]):
+        if config["MEDIA_DIR"]:
+            if not os.path.isdir(config["MEDIA_DIR"]):
+                os.makedirs(config["MEDIA_DIR"])
+        if not os.path.isdir(config["MEDIA_DIR"]):
+            config["MEDIA_DIR"] = "./media"
         else:
             print(
```

Comment:
You didn't add this, but this should be logged rather than printed.

Reply:
Changed in the latest commit.